controllers/hetznercluster_controller_test.go

```diff
@@ -840,3 +842,100 @@ var _ = Describe("reconcileRateLimit", func() {
 		Expect(reconcileRateLimit(hetznerCluster, testEnv.RateLimitWaitTime)).To(BeFalse())
 	})
 })
+
+func TestSetControlPlaneEndpoint(t *testing.T) {
+	t.Run("function should return false if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is not enabled and hetznerCluster.Spec.ControlPlaneEndpoint is nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: nil}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == true){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint != nil){
+			t.Fatalf("Wrong value for hetznerCluster.Spec.ControlPlaneEndpoint. Got: %s, Want: <nil>", hetznerCluster.Spec.ControlPlaneEndpoint)
+		}
+		assert.Equal(t, false, got)
+	})
+	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is false but hetznerCluster.Spec.ControlPlaneEndpoint is not nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: &clusterv1.APIEndpoint{Host: "xyz"}}}
+		got := SetControlPlaneEndpoint(hetznerCluster)
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled != false){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint == nil){
+			t.Fatalf("ControlPlaneEndpoint must not be nil")
+		}
+		if (hetznerCluster.Spec.ControlPlaneEndpoint.Host != "xyz"){
```

Comment:
you also should test here that the port is still not set... otherwise we cannot be sure that no changes have been made to the port